packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -262,9 +376,56 @@ export const prepareStakingAccountKit = (baggage, makeRecorderKit, zcf) => {
           if (!balance) throw Fail`Result lacked balance key: ${result}`;
           return harden(toChainAmount(balance));
         },
+
+        withdrawRewards() {
+          throw assert.error('Not implemented');
+        },
+
+        /**
+         * @param {Delegation[]} delegations
+         */
+        async undelegate(delegations) {
+          trace('undelegate', delegations);
+          const { helper } = this.facets;
+          const { chainAddress, bondDenom, timer } = this.state;
+
+          const result = await E(helper.owned()).executeEncodedTx(
+            delegations.map(d =>
+              toAnyJSON(
+                MsgUndelegate.toProtoMsg({
+                  delegatorAddress: chainAddress.address,
+                  validatorAddress: d.validatorAddress,
+                  amount: { denom: bondDenom, amount: d.shares },
+                }),
+              ),
+            ),
+          );
+
+          const response = tryDecodeResponse(
+            result,
+            MsgUndelegateResponse.fromProtoMsg,
+          );
+          trace('undelegate response', response);
+          const { completionTime } = response;
+          const endTime = BigInt(completionTime.valueOf() / 1000);
```

Comment:
Is `completionTime` a `Date` object? Came here to see if there was any nuance in handling this in Jessie

Reply:
> Is `completionTime` a `Date` object?

yes. I suppose `getTime()` would be more clear than `valueOf()`.

> Came here to see if there was any nuance in handling this in Jessie

only in constructing them, which our code doesn't do.